lib/model.js

```diff
@@ -362,6 +362,14 @@ ModelBaseClass.prototype._initProperties = function(data, options) {
   this.trigger('initialize');
 };
 
+// Helper function for determing the applyDefaultOnWrites value of a property
+function applyDefaultOnWrites(property) {
```

Comment:
The name "applyDefaultOnWrites" implies to me that the function will apply those default values when appropriate, which is not true. Let's use a more accurate name please, for example `shouldApplyDefaultsOnWrites` or `appliesDefaultsOnWrites`.